### PR TITLE
fix #73907   Stop Flip Flopping mutations in Alpha stomach chain

### LINKED ISSUE #73907: Weak stomach/nausea/vomitous trait keeps looping itself into upgrading and downgrading.
### Describe the bug

If you take mutagen catalyst and rat mutagen and get basically every mutation at some point, you will loop between the weak stomach/nausea/vomitous traitline.

### Attach save file

[Ridge Ratta-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/15366096/Ridge.Ratta-trimmed.tar.gz)

### Steps to reproduce

1.Take tons of rat mutagen
2. Wait to get mutations
3. Watch weak stomach loops back between nauseous and vomitous

### Expected behavior

I assume it isn't supposed to be upgrading and downgrading itself
### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.4412 (22H2)
- Game Version: cdda-experimental-2024-05-13-1412 dd42dbb [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Extra Lives - Extended [extra_lives_extended],
    Bionic Slots [cbm_slots],
    Extra Mutated Scenarios [extra_mut_scens],
    Mind Over Matter [mindovermatter]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kamejeir**
#### Summary
Bugfixes "Flip-flopping in acquisition of Intestinal Fortitude"

#### Purpose of change

bugfix #73907

#### Describe the solution

Weak Stomach now leads to Nausea, that itself turns into Vomitous.

Intestinal Fortitude has its different prereqs replaced with Weak Stomach and Vomitous, the two of which are set to change into Intestinal Fortitude.

Removes Robust Genetics from Alpha.  Alpha should be a jealous category.

#### Testing

It only go in one direction, and through a couple runs of mutating Rat, never observed Vomitous turning into Nausea.

Weak Stomach, Nausea and Vomitous successfully disappear after acquiring Intestinal Fortitude.

### CODE REVIEW

**Comment by Kamejeir** [comment]
Tests are failing because breach power is too high. I could shove something else around, or edit the test to make Alpha slightly easier to get. I don't know which to choose.

**Comment by Kamejeir** [comment]
Now after Venera telling me that I am not to change the test, the other option has me wondering what to even do. I haven't ever gotten involved with the balance of Alpha. My best and only guess is to move Deterioration to post-thresh.

**Review by Maleclypse — CHANGES_REQUESTED** [review]
Do not merge until I can look at this.

**Comment by Maleclypse** [comment]
Deterioration to post thresh is not going to work. I’ll look at some options. I think in this specific instance Venera may be wrong about the test because it’s the alpha category and if he isn’t wrong it should be a positive mutation that moves to post thresh.

**Comment by Kamejeir** [comment]
Actually, looking back, it is extremely wacky that the test succeeded. The total amount of mutations potentially available to breach didn't change (Weakening just turned into Deterioration, at no point were there more Alpha mutations to go post-threshold). It makes me think the test is unreliable and should be changed.

**Comment by Kamejeir** [comment]
These ones could use a 61 instead of 60 in case it is determined that the test should be changed rather than editing Alpha stuff.
https://github.com/CleverRaven/Cataclysm-DDA/blob/8b1645f8c0a873911d0639f235511c5fb404dc6d/tests/mutation_test.cpp#L213
https://github.com/CleverRaven/Cataclysm-DDA/blob/8b1645f8c0a873911d0639f235511c5fb404dc6d/tests/mutation_test.cpp#L243-L246

**Comment by Kamejeir** [comment]
btw are you on github via the site or..? edited the first message after your actions a lot, it changed meaning pretty drastically, dunno if edits are re-sent as a new message

**Comment by Maleclypse** [comment]
> btw are you on github via the site or..? edited the first message after your actions a lot, it changed meaning pretty drastically, dunno if edits are re-sent as a new message

They aren’t sent as new messages. No worries. So the issue with the test is that alpha has a unique test in the larger threshbreak test because it has fewer mutation than you can breach with otherwise. It’s the smallest tree. The note on the test is wrong but basically because it’s so small you have to change the threshbreach chance if you add a mutation or else it will be unnaturally easy to become thresh alpha

**Comment by Kamejeir** [comment]
Then it appears as though we have to move an independent mutation (one that does not change to anything else / is a parent that something else changes into, and thus actually makes a difference for breach power) post-threshold.
The three choices that fits the bill are as follows:
- Weak Scent (starting trait quality would probably have to go)
- Picky Eater (ditto for above, as well as this believably imo being what an "average dude" could have)
- Highly Vain

File: data/json/mutations/mutations.json
Comment (Maleclypse) [inline data/json/mutations/mutations.json:6998]:
```suggestion
```

**Comment by Kamejeir** [comment]
Well, color me impressed. I had failed to understand that it was about points within the mutations, not the number of mutations.

**Comment by Kamejeir** [comment]
Oops, too hasty a judgment. Well, that was a failure.

File: data/json/mutations/mutations.json
Comment (Maleclypse) [inline data/json/mutations/mutations.json:6991]:
```suggestion
```

**Comment by Maleclypse** [comment]
> Oops, too hasty a judgment. Well, that was a failure.

Yeah I expected that would likely be a failure because I was pretty sure negatives didn't count by points.  Other people seemed to think this might be a way to impact points by adjusting the positives again, not sure that the points the test is counting correlate.

**Comment by Maleclypse** [comment]
```
 if( !have_upgrade && !mdata.flags.count( json_flag_NON_THRESH ) ) {
            // ... find the mutations distance from normalcy...
            int mut_height = mutation_height( mut );
            // ... and for each category it falls in...
            for( const mutation_category_id &cat : mdata.category ) {
                // ... add the height and a constant value
                mutation_category_level[cat] += mut_height + 2;
```  So by making the lead to instead of change to you've added one to the height of the stomach branch.  So none of these changes I've been trying have a chance of working.   I'm going to make a PR to remove the special treatment for Alpha threshbreach and after that is merged this will pass.

**Review by Maleclypse — APPROVED** [review]

**Pull request merged**
